src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -279,10 +280,8 @@ public class PullRequestHandler : IHttpHandler
 
         string[] gitCommands = new string[]
         {
-            "checkout -f master",
-            "clean -fdx",
             "fetch --all --prune",
-            "merge --ff-only origin/master"
+            "checkout -f origin/master"
```

Comment:
>checkout -f origin/master [](start = 13, length = 25)

Non-blocking question: could you help me understand exactly what git does in the new code you've added?  For example, I see that a forced checkout to origin/master puts the repro into a state with a detached HEAD.  Then it deletes *all* branches, including master, and creates a new tracking branch called 'master' that tracks origin/master and makes that the current branch.  So the very first time this runs (after someone did the first clone), it deletes the "normal" local master branch.  Subsequent runs delete the tracking branch called master.  In all cases, it works with a pristine copy of what is in origin/master that does not depend on the prior state of the master.  Do I have that right?